src/polyfill.ts

```diff
@@ -83,17 +86,34 @@ export class LockManager {
       ifAvailable: false,
       steal: false,
     };
-    this._clientId = generateRandomId();
+    this._clientId = `${LIB_PREFIX}-clientId-${generateRandomId()}`;
     this._init();
   }
 
   private _init() {
+    this._storeThisClientId();
     const heartBeat = new HeartBeat({ key: this._clientId });
     heartBeat.start();
+    // handle when unload could't work or the client crash, then clean
     heartBeat.detect(() => this._cleanUnliveClientLocks());
     this._onUnload();
   }
 
+  private _getClientIds(): string[] {
+    const clientIds = getStorageItem(STORAGE_KEYS.CLIENT_IDS);
+    return (clientIds && JSON.parse(clientIds)) || [];
+  }
+
+  private _storeClientIds(clientIds: string[]) {
+    setStorageItem(STORAGE_KEYS.CLIENT_IDS, JSON.stringify(clientIds));
+  }
+
+  private _storeThisClientId() {
+    const prevClientIds = this._getClientIds();
+    const curClientIds = [...prevClientIds, this._clientId];
+    this._storeClientIds(curClientIds);
+  }
```

Comment:
What if a different tab updates the client IDs item in local storage in-between _getClientIds() and _storeClientIds()?

Reply:
then _storeClientIds will store the new client ID, then _getClientIds will get all the IDs including old, new  client IDs, and _cleanUnliveClientLocks will clean the old client ID by heartbeat detect

Reply:
If two clients each run _storeThisClientId() at the same time, you might end up with the case where both clients read clientIds=[], then the first will try to save clientIds=[Client1] and the second will try to save clientIds=[Client2]. Only one will win. Example:

1. Client 1 reads that clientIds = []
2. Client 2 reads that clientIds = []
3. Client 1 saves clientIds = [Client1]
4. Client 2 saved clientIds = [Client2]
5. Local storage now has only clientIds = [Client2]

Looking around the code, I think _cleanClientLocksByClientId also assumes that no other clients, will be modifying local storage in between reading and writing?

Reply:
I have tested the storage reading and writing operation spending time, It is very fast, millisecond-level. In my testing, do  `getItem`  `JSON.Parse` `JSON.Stringfy` `setItem ` , which for reading and writing, complete among 1 millisecond.
So when we reopen the tabs, it is not the real millisecond-level same time.

Reply:
Got it. I found this library looking for away to ensure my example case is impossible, but guess it wasn't the goal of this this polyfill.

Reply:
The only case I met is, inside this lib, the shared lock released in the real same time among two clients, and modifying the same one local storage, then got the issue you remind.  You could see the handle method `_handleSharedLockFromListener `.